### ISSUE TITLE
Address possible break of project installation after retry install due to certain incompatible framework

### ISSUE THREAD

**Issue by erdembayar** [NuGet/Home#9944]
## Details about Problem
Worked before? N/A
During my performance improvement work [Nuget/Home#6010](https://github.com/NuGet/Home/issues/6010) Nikolche [pointed out](https://github.com/NuGet/NuGet.Client/pull/3559/files#r475033357) potential bug where initially success installation package break due to re-evaluation. When child project re-evaluated then it might cause changes in parent project of this project which in turn might break. So I created this follow up issue for investigate his comment. 

#### Update
My conclusion for this investigation is it's not breaking the parent project when we update child project due to certain incompatible framework.

## Detailed repro steps so we can see the same problem
N/A : Not sure it's just possible bug.

**Comment by erdembayar** [comment]
@nkolev92 
It looks things are working as expected for your concern when I test 2-3 scenarios.
1. Here is example which you mentioned last time. I upgraded to new version of package version which targets different framework.
![image](https://user-images.githubusercontent.com/8766776/92658367-4dfd5980-f2ab-11ea-96c7-613fe9f8e5f1.png)

Here is how it works gif animation:
![PackageCompatibilityCheck](https://user-images.githubusercontent.com/8766776/92658451-79804400-f2ab-11ea-8ad3-3d7810737b9a.gif)

It even doesn't get inside condition where checking because unsuccessfullFramework.Count == 0.
![image](https://user-images.githubusercontent.com/8766776/92664003-adf9fd00-f2b7-11ea-8293-b47255de630b.png)

2. I think it would be problem only for install the package first time, not upgrade as above.

Because check here:
https://github.com/NuGet/NuGet.Client/blob/9f9870a6b47f7918524d68e52ce21c422ce1af12/src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs#L2820-L2821

Here is I tried first time install package which goes inside [this method](https://github.com/NuGet/NuGet.Client/blob/9f9870a6b47f7918524d68e52ce21c422ce1af12/src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs#L2823), but after restore parent project are looks still good.
I installed different combination of frameworks to check if any them fails.
Please note here: `ProjClassLibrary2` is child project so any change in it might cause changes in parents `ConsoleApp5` and `ConsoleCoreApp1`.

2. Here I install package to both parents and a child project.
a.
 ![Ver1](https://user-images.githubusercontent.com/8766776/92674494-4ac99400-f2d2-11ea-808c-17ae8c367b93.gif)
b. 
![Ver2](https://user-images.githubusercontent.com/8766776/92674663-a72cb380-f2d2-11ea-8a18-0b0c09d77112.gif)
c.
![Ver3](https://user-images.githubusercontent.com/8766776/92674702-b7dd2980-f2d2-11ea-906e-55d528fae17e.gif)

d.
![Ver4](https://user-images.githubusercontent.com/8766776/92674715-bd3a7400-f2d2-11ea-82c8-e7049b527109.gif)

3. Here I install package to child project lower  version than parent projects, and it looks parent project not affected by this change.
![InstallToChildProjOnly1](https://user-images.githubusercontent.com/8766776/92682114-59b94200-f2e4-11ea-9c49-f98910cbec0f.gif)

4. Here I install package to child project higher version than parent projects, and it looks parent project not affected by this change.
![InstallTOChildProjOnly2](https://user-images.githubusercontent.com/8766776/92682121-5d4cc900-f2e4-11ea-9cdc-96f0588c0bcc.gif)

If you wish I can add unit tests for above cases, but not sure if it's necessary.

**Comment by erdembayar** [comment]
@nkolev92  please review this.

**Comment by nkolev92** [comment]
@erdembayar The gif is different from your test description.

Your test has Project 1 with the net472;netcoreapp3.0, but the description has net46;netcoreapp3.0. 

The description seems like the correct one. 

Probably best to re-run the tests with the corrected test case.

**Comment by erdembayar** [comment]
@nkolev92 
I re-tested with exact version as described.
Project1 : netcoreapp3.0;net46
Project2 : netcoreapp3.0
Project3 : netcoreapp3.0
PackageA: V1 net45, .netstandard16

Upgrade to PackageA V2: .netstandart20

1. First I tried to upgrade PackageA to Project1, but it failed due to `NU1202`. But it didn't corrupt previously valid state, simple it just didn't upgrade at all, so nothing changed. 	`Package PackageA 2.0.0 is not compatible with net46 (.NETFramework,Version=v4.6). Package PackageA 2.0.0 supports: netstandard2.0 (.NETStandard,Version=v2.0)`	C:\Users\eryondon\source\repos\CompatibilityTest\Project1\Project1.csproj	1	

![Proj1](https://user-images.githubusercontent.com/8766776/94770246-f9f50a80-0368-11eb-8cb4-fad577f8ee9d.gif)

2. Then I tried to upgrade PackageA to Project2 and Project 3, then both of them worked.

![ProjParents](https://user-images.githubusercontent.com/8766776/94770304-1ee97d80-0369-11eb-8481-9f9788d6e10e.gif)

Again I think it would be problem only for install the package first time, not upgrade as above one. 
For first time install case:
PackageA V1.00 installs without any problem to all project.
But PackageA V2.00 install fails same reason as above. `NU1202`
